ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/fastsync/WorldStateDownloader.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.sync.fastsync;
+
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface WorldStateDownloader {
+  CompletableFuture<Void> downloadWorldState(BlockHeader pivotBlockHeader);
```

Comment:
@mbaxter this is the API I'd ideally like for the world state downloader which doesn't quite match #657 where the pivot block header is passed into the constructor.  There are two ways around this which both seem reasonable:

1. Make this interface act as a factory that creates a new instance of the world state downloader as it exists
2. Change the world state downloader run method to take the pivot block header

The main thing is to ensure we don't wind up with two concurrent world state downloads which is maybe slightly more straight forward with option 2, but the way `FastSyncDownloader` manages the returned futures should take care of it anyway.

Reply:
I like option #2 - that probably make more sense than fixing the header in the constructor.